### ISSUE TITLE
[Enhancement] Groups: Allow group admins to assign existing users

### ISSUE THREAD

**Issue by TomTurnschuh** [nextcloud/server#11334]
Group administrators can add and delete users in the group they have admin privileges for. This can lead to multiple users for the same person, as a group admin creating a new user may not know that there already exists such a user – and if he knows, he cannot assign it to his group.

I would suggest that group administrators can only or additionally assign existing users to the group they have admin privileges for.
For this group administrator needs to access the list of all users. However, the visible information can be limited to user and display name and potentially group memberships.

**Comment by TomTurnschuh** [comment]
This proposal is related to (but not duplicate of) #7482.

**Comment by VincentWo** [comment]
Hi is there anything actually speaking against adding this feature? Or does it just need to be implemented?

**Comment by WingsLikeEagles** [comment]
This is a very important feature.  What's the purpose of having a group admin if they can't change group membership?

**Comment by Tash-Feneko** [comment]
This is crucial feature for group admins to actually work, is there any chance of it being implemented in any foreseeable future?

**Comment by hansimausisteffiwurstbrot** [comment]
In our case I don't want group admins to be able to add any existing user to their group due to privacy reasons.

Might be handled as a configurable option?

**Comment by synlic** [comment]
I agree this i really needed. I've setup a Nextcloud for a local non-profit with several smaller organisations, and it's not really feasable for only the site admins to add users to groups. We have a work around of not using groups and only Circles, but not all apps have full support for Circles.

**Comment by ZeikoFr** [comment]
This feature is clearly needed.

I think the best thing to do is : 

Allow group admin to add only user that are already existing in the Nextcloud.

But we should also have an option to restrict this option to some user like a button "Can be added to other group by group admin" To avoid human mistakes (Like HR group admin adding a user that is not related to HR in is group).

**Comment by WingsLikeEagles** [comment]
Ping, would be helpful to see this feature added.

**Comment by tech-nickel** [comment]
Same for us. I think it's not the way it should be to allow Group admins to add completely new users to the instance on a global level. Only existing user should be available to add to the group.

**Comment by geonom** [comment]
Can only support this, would be greatly useful to us as a group with about 300 users at least, if group admins could help with adding and removing existing users from groups on their own without needing a general admins account with all the dangerous priviledges.

**Comment by walhallaRV** [comment]
> Group administrators can add and delete users in the group they have admin privileges for. This can lead to multiple users for the same person, as a group admin creating a new user may not know that there already exists such a user – and if he knows, he cannot assign it to his group.
> 
> I would suggest that group administrators can only or additionally assign existing users to the group they have admin privileges for. For this group administrator needs to access the list of all users. However, the visible information can be limited to user and display name and potentially group memberships.

Yes, I agree. Absolutely necessary / essential for bigger Groupware / Workgroup - Servers. May be that there should be the option do deactivate for privacy reasons. I agree too. But as it is, it is really pain in the ...

Group admin can't create new groups, nor add (existing) users to the group. So he has to borrow the Super Admin like: "Oh, please can you create a new group named "Test" and add this 250 users to the group, please!" GREAT! 

But really the best: the Superadmin can not simply add the users to the group. NO, he has to edit every single user, has to add 250 times manually the group to the 250 users! Really a pity and not usable for greater groups :(

**Comment by sorbaugh** [comment]
Descoped from NC32

**Comment by aparcar** [comment]
How come this was descoped again?
